subsys/shell/shell_utils.c

```diff
@@ -547,6 +554,24 @@ const struct device *shell_device_get_binding(const char *name)
 	return dev;
 }
 
+const struct device *shell_device_get_binding_all(const char *name)
+{
+	const struct device *dev;
+	size_t len = z_device_get_all_static(&dev);
+	const struct device *dev_end = dev + len;
+
+	while (dev < dev_end) {
+		if ((dev->name != NULL) &&
```

Comment:
```suggestion
		if (dev->name &&
```
Isn't this the common approach in zephyr as well when checking non-null?

Reply:
I c/p shell_device_internal().
In linux it is the way yes (like testing any integer if it has a value or 0) but in zephyr... I thought there might be some MISRA rule for this, but looking at https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html I am not so sure anymore (could not find anything related to this)